File: java/dsa/algorithms/strings/count-and-say.java
public String countAndSay(int n) {
  String sequence = "1";
  for (int i = 1; i < n; i++) {
    StringBuilder nextSequence = new StringBuilder();
    for (int j = 0; j < sequence.length(); j++) {
      int count = 1;
      while (j + 1 < sequence.length() && 
             sequence.charAt(j) == sequence.charAt(j+1)) {
        count++;
        j++;
      }
      nextSequence.append(count).append(sequence.charAt(j));
    }
    sequence = nextSequence.toString();
  }
  return sequence;
}

File: go/dsa/algorithms/strings/count-and-say.go
func countAndSay(n int) string {
  sequence := "1"
  for i := 1; i < n; i++ {
    var nextSequence strings.Builder
    for j := 0; j < len(sequence); {
      count := 1
      for j+1 < len(sequence) && sequence[j] == sequence[j+1] {
        count++
        j++
      }
      nextSequence.WriteString(strconv.Itoa(count))
      nextSequence.WriteByte(sequence[j])
      j++
    }
    sequence = nextSequence.String()
  }
  return sequence
}